app/services/applier/form_filler.rb

```diff
@@ -1,69 +1,118 @@
 # frozen_string_literal: true
 
-require 'json'
-# require 'open-uri'
-# require 'htmltoword'
-
-# TODO: Handle job posting becoming closed (redirect or notification on page)
 module Applier
-  # Core class for filling out forms using Capybara
   class FormFiller < ApplicationTask
     include Capybara::DSL
-
-    def initialize(job_application, payload)
-      @job_application = job_application
-      @url = @job_application.job.posting_url
-      @ats = @job_application.job.applicant_tracking_system
-      @fields = JSON.parse(payload)
-      # @user = @job_application.application_process.user
-      @errors = nil
-
-      # include the relevant ATS form_filler module
-      include_ats_module
+    def initialize(payload)
+      @application_form = payload[:form_locator]
+      @fields = payload[:fields]
+      @session = Capybara::Session.new(:selenium)
+      @url = payload[:apply_url]
+      @user_fullname = payload[:user_fullname]
     end
 
     def call
-      return unless processable
+      return false unless processable
 
-      process
-    rescue StandardError => e
-      Rails.logger.error "Error running FormFiller: #{e.message}"
-      nil
+      Capybara.using_session(@session) do
+        process
+      end
     end
 
     private
 
     def processable
-      @url && @fields && @job_application && @ats
+      @url && @fields && @session
     end
 
     def process
-      p "Hello from FormFiller!"
-      submit_application
+      visit_url
+      click_apply_button
+      fill_in_all_fields
+      sleep 2
+      click_submit_button
+      confirm_submission_was_successful
+    ensure
+      @session.quit
+    end
```

Comment:
What about something like this instead ?

```
def process
  visit_url
  fill_application_form
  click_submit_button
  # sleep 2 add this to click_submit_button
  verify_submission # instead of confirm_submission_was_successful
ensure
  @session.quit
end

private

def fill_application_form
  click_apply_button
  fill_in_all_fields
end

def verify_submission
  # Add logic to check for successful submission message or other indicators
end
```

Reply:
Thanks! :)